pallets/automation-time/src/lib.rs

```diff
@@ -251,5 +292,56 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Run tasks for the block time.
+		///
+		/// Complete as many tasks as possible given the maximum weight.
+		/// Return the weight that was used.
+		///
+		/// Until the TODO is completed we will be limited to two tasks per slot to ensure
+		/// they can all be completed.
+		///
+		/// TODO (ENG-157):
+		/// - Calculate weights for each task
+		pub fn run_tasks(max_weight: Weight) -> Weight {
+			let mut weight_left: Weight = max_weight - RUN_TASK_OVERHEAD;
+			let time_block = Self::get_current_time_slot();
+
+			if let Some(task_ids) = Self::get_scheduled_tasks(time_block) {
+				let mut consumed_task_index: usize = 0;
+				for task_id in task_ids.iter() {
+					if weight_left < MAX_LOOP_WEIGHT {
+						break
+					}
+
+					let cost = match Self::get_task(task_id) {
+						None => 0, // TODO: add some sort of error reporter here (ENG-155).
+						Some(task) => match task.action {
+							Action::Notify(message) => Self::run_notify_task(message),
+						},
+					};
+
+					consumed_task_index += 1;
+					weight_left = weight_left - cost - RUN_TASK_LOOP_OVERHEAD;
+				}
+
+				if consumed_task_index == task_ids.len() {
+					<ScheduledTasks<T>>::remove(time_block);
+				} else {
+					let new_task_ids: BoundedVec<T::Hash, T::MaxTasksPerSlot> =
+						task_ids.into_inner().split_off(consumed_task_index).try_into().unwrap();
+					<ScheduledTasks<T>>::insert(time_block, new_task_ids);
+				}
+			}
+
+			max_weight - weight_left
+		}
+
+		/// Fire the notify event with the custom message.
+		/// TODO: Calculate weight (ENG-157).
+		fn run_notify_task(message: Vec<u8>) -> Weight {
+			Self::deposit_event(Event::Notify { message });
```

Comment:
this goes tot he treasury right?

Reply:
I am not sure what you mean. This sends an event. It does not move tokens around or anything

Reply:
there's also a depost funds event for transaction fees, i think; i was confusing that w/ this